File: src/main/java/com/freya02/botcommands/annotation/RequireOwner.java
package com.freya02.botcommands.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/** Use this annotation if you want the command to be usable only by the owners, <b>the 'help' command will still display it</b>
 *
 * <br>
 * <h2>This does not work on slash commands</h2>
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface RequireOwner {}

File: BotCommands-commands/text/src/main/kotlin/io/github/freya022/botcommands/api/commands/text/annotations/RequireOwner.kt
package io.github.freya022.botcommands.api.commands.text.annotations

import io.github.freya022.botcommands.api.commands.text.builder.TextCommandBuilder
import io.github.freya022.botcommands.api.core.BotOwners

/**
 * Marks this text command as only usable by the [bot owners][BotOwners].
 *
 * Owner-only commands are hidden in the built-in help content,
 * but will still be responded to if a user tries to use it,
 * though they will be rejected.
 *
 * ### Annotation scope
 * This annotation applies to the command itself, not just on a single variation,
 * in other words, it applies to all commands sharing the same path.
 *
 * @see TextCommandBuilder.ownerRequired DSL equivalent
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class RequireOwner
